File: 15주차/KM/111-minimum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int min = Integer.MAX_VALUE;
    public int minDepth(TreeNode root) {
        if(root == null) return 0;
        dfs(root, 1); 
        return min;
    }
    public void dfs(TreeNode node, int depth){
        if(node == null) {
            return;
        }
        if(node.left == null && node.right == null){
            min = Math.min(depth, min);
        }
        dfs(node.left, depth+1);
        dfs(node.right, depth+1);
    }
}

File: 15주차/DE/111. Minimum Depth of Binary Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
 function minDepth(root) {
    if(!root) return 0;
    
    if(!root.left && !root.right) {
        return 1;
    }
    
    let left = minDepth(root.left) + 1;
    let right = minDepth(root.right) + 1;
    
    if(!root.left) return right;
    if(!root.right) return left;

    return Math.min(left, right);
};